apps/core/lib/core/models/user.ex

```diff
@@ -41,14 +41,20 @@ defmodule Core.User do
     model
     |> __MODULE__.changeset(params)
     |> cast(params, [:email, :password, :role])
-    |> validate_required([:email, :password, :role])
+    |> validate_required([:email, :role])
     |> validate_inclusion(:role, @valid_roles)
     |> validate_format(:email, @valid_email_format)
     |> unique_constraint(:email)
     |> validate_length(:password, @valid_password_length)
+    |> validate_confirmation(:password, message: "does not match password!")
     |> encrypt_passwd()
   end
 
+  def required_password_changeset(model, params) do
```

Comment:
`validate_confirmation` should be in this changeset, and this changeset, as we discussed previously, should be called `password_changeset`